src/javascripts/ng-admin/Crud/delete/DeleteController.js

```diff
@@ -22,14 +22,11 @@ define(function () {
 
     DeleteController.prototype.deleteOne = function () {
         var notification = this.notification,
-            $state = this.$state, entityName = this.entity.name();
+            entityName = this.entity.name(),
+            $window = this.$window;
 
         this.WriteQueries.deleteOne(this.view, this.entityId).then(function () {
-
-            $state.go($state.get('list'), angular.extend({
-                entity: entityName,
-                id: this.entityId
-            }, $state.params));
+            this.back();
```

Comment:
@ahgittin: incoming #642 indeed fixes this issue. If you are on the show view and click delete, the controller is going to check if the previous page is related to the deleted entity. If so, it redirects you to the list view. Otherwise, to the previous page.